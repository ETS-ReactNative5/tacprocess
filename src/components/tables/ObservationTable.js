import React from 'react'
import propTypes from 'prop-types'
import '../../styles/components/tables.css'
import ObservationTableRow from './tablesComponents/ObservationTableRow'

const ObservationTable = ({priorityObservations}) => {
  if (!priorityObservations.length) {
    return null
  }
  return (
    <table style={ { width: '100%' } }>
      <thead>
        <tr>
          <th>Priority</th>
          <th>Percentage</th>
        </tr>
      </thead>
      <tbody>
        {
          priorityObservations.map(o => (
            <ObservationTableRow
              key={ o.priority }
              observation={ Object.keys(o).length ? o : { } }
            />))
        }
      </tbody>
    </table>
  )
}

ObservationTable.propTypes = {
  priorityObservations: propTypes.array.isRequired
}
export default ObservationTable
