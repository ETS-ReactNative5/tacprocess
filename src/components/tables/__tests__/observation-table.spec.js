import React from "react"
import { mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import ObservationTable from "../ObservationTable"

const proposals = [
  { proposalCode: 'Code-1',
    title: 'Title-1',
    principalInvestigator: { givenName: 'PI-1', familyName: 'PI-1'},
    liaisonAstronomer: { givenName: 'LA-1', familyName: 'LA-1'},
    status: 'Status-1',
    statusComment: 'Comment-1',
    completionComments: [{ semester: '2018-1', comment: 'Testing'}],
    timeAllocations: [{ priority: 1, amount: 50000, semester: '2018-1', partnerCode: 'RSA' }],
    observations: [{ status: 'ACCEPTED', block: { length: 2000, priority: 1, semester: '2018-1'} }]
  }
]

// Checking if the ObservationTable Component renders correctly for different input
describe("ObservationTable Component", () => {
  
  //Enzyme method testing if it renders correctly with empty values of the proposals
  it("Render the ObservationTable Component having unpopulated props with no errors", () => {
    const rendered = mount(<ObservationTable
      proposals= {[]}
      semester={''}
      partner={''}
    />)
    expect(shallowToJson(rendered)).toMatchSnapshot()
  })
  
  it("Render the ObservationTable Component having populated props with no errors", () => {
    const rendered = mount(<ObservationTable
      proposals= {proposals}
      semester={'2018-1'}
      partner={'RSA'}
    />)
    expect(shallowToJson(rendered)).toMatchSnapshot()
  })
})
