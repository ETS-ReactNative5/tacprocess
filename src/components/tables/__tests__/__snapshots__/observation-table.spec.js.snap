// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationTable Component Render the ObservationTable Component having populated props with no errors 1`] = `
<ObservationTable
  partner="RSA"
  proposals={
    Array [
      Object {
        "completionComments": Array [
          Object {
            "comment": "Testing",
            "semester": "2018-1",
          },
        ],
        "liaisonAstronomer": Object {
          "familyName": "LA-1",
          "givenName": "LA-1",
        },
        "observations": Array [
          Object {
            "block": Object {
              "length": 2000,
              "priority": 1,
              "semester": "2018-1",
            },
            "status": "ACCEPTED",
          },
        ],
        "principalInvestigator": Object {
          "familyName": "PI-1",
          "givenName": "PI-1",
        },
        "proposalCode": "Code-1",
        "status": "Status-1",
        "statusComment": "Comment-1",
        "timeAllocations": Array [
          Object {
            "amount": 50000,
            "partnerCode": "RSA",
            "priority": 1,
            "semester": "2018-1",
          },
        ],
        "title": "Title-1",
      },
    ]
  }
  semester="2018-1"
/>
`;

exports[`ObservationTable Component Render the ObservationTable Component having unpopulated props with no errors 1`] = `
<ObservationTable
  partner=""
  proposals={Array []}
  semester=""
/>
`;
